Flagsmith.EngineTest/Unit/Segments/SegmentEvaluatorTest.cs

```diff
@@ -173,6 +173,29 @@ public void TestIdentityInSegmentPercentageSplit(int segmentSplitValue, int iden
             Evaluator.Hashing = hashingMock.Object;
             var result = Evaluator.EvaluateIdentityInSegment(Unit.Fixtures.Identity(), segment, null);
             Assert.Equal(expectedResult, result);
+            Evaluator.Hashing = new Hashing();
         }
+        [Theory]
+        [MemberData(nameof(TestCasesIdentities))]
+        public void TestIdentityInSegmentPercentageSplitUsesDjangoId(IdentityModel identity, bool expectedResult)
+        {
+            var percentage_split_condition = new SegmentConditionModel
+            {
+                Operator = Constants.PercentageSplit,
+                Value = "50",
+            };
+            var rule = new SegmentRuleModel { Type = Constants.AllRule, Conditions = new List<SegmentConditionModel> { percentage_split_condition } };
+            var segment = new SegmentModel { Id = 1, Name = "% split", Rules = new List<SegmentRuleModel> { rule } };
+
+            var result = Evaluator.EvaluateIdentityInSegment(identity, segment, null);
+
+            Assert.Equal(expectedResult, result);
+        }
+        public static IEnumerable<object[]> TestCasesIdentities() =>
+            new List<object[]>
+            {
+                new object[]{new IdentityModel(){ Identifier = "noDjangoID", EnvironmentApiKey = "key" }, true},
+                new object[]{new IdentityModel(){ DjangoId = 1, Identifier = "noDjangoID", EnvironmentApiKey = "key" }, false},
```

Comment:
```suggestion
                new object[]{new IdentityModel(){ DjangoId = 1, Identifier = "withDjangoID", EnvironmentApiKey = "key" }, false},
```

Reply:
The identifiers are supposed to be identical here.

Reply:
Discussed in more detail here: https://github.com/Flagsmith/flagsmith-go-client/pull/88#discussion_r1230882061

Reply:
Changed to "Test".